From 084390bc521a4e3f1241b110714dbc73ac58859a Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Fri, 26 Jun 2020 17:51:56 +0200
Subject: [PATCH] PONRTSYS-7705: Set color marking to internal in delete

When the color marking is deleted in the PON QoS driver, set it to
internal and not green marking as before.
---
 drivers/net/datapath/pon_qos/pon_qos_tc_colmark.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_colmark.c b/drivers/net/datapath/pon_qos/pon_qos_tc_colmark.c
index fb84a1dcb6cd..751b883ea33b 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_colmark.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_colmark.c
@@ -124,7 +124,7 @@ int pon_qos_colmark_unoffload(struct net_device *dev,
 	 * dp_meter_add() or dp_meter_del() change.
 	 */
 	meter_cfg->col_mode = 0;
-	meter_cfg->mode = DP_NO_MARKING;
+	meter_cfg->mode = DP_INTERNAL;
 	meter_cfg->type = srTCM;
 	ret = dp_meter_add(dev, meter_cfg, flags);
 	if (ret != DP_SUCCESS) {
